spec/filters/geoip_spec.rb

```diff
@@ -184,15 +185,46 @@
 
     context "should not raise an error" do
       sample("ip" => "-") do
-        expect{
-          subject
-          }.to_not raise_error
+        expect{ subject }.to_not raise_error
       end
 
       sample("ip" => "~") do
-        expect{
-          subject
-          }.to_not raise_error
+        expect{ subject }.to_not raise_error
+      end
+    end
+
+    context "when IO is used for logging" do
```

Comment:
looking at this test I'm not sure if we should try to work with whatever `@logger.error` gives us and ensure that either we get a proper Struct from `get_geo_data` or consider the call failed.
This way we don't have to go into details like "what if Cabin logger uses a StringIO?"